hazelcast/src/main/java/com/hazelcast/spi/impl/BasicOperationService.java

```diff
@@ -1090,6 +1101,8 @@ private void assertNoBackupOnPrimaryMember(InternalPartition partition, Address
 
         public static final int DELAY_MILLIS = 1000;
 
+        private volatile boolean shutdown;
```

Comment:
Problem with relying on interrupt flag is that the interrupt could have been consumed somewhere and then you don't see it. This is imho more reliable.